src/codecs/avif/decoder.rs

```diff
@@ -172,3 +172,30 @@ impl<'a, R: 'a + Read> ImageDecoder<'a> for AvifDecoder<R> {
         Ok(())
     }
 }
+
+/// `get_picture` and `send_pending_data` yield `Again` as a non-fatal error requesting more data is sent to the decoder
+/// This ensures that in the case of `Again` all pending data is submitted
+/// This should be called after `send_data` (which does not yield `Again` when called the first time)
+fn read_until_ready(decoder: &mut dav1d::Decoder) -> ImageResult<dav1d::Picture> {
+    loop {
+        match decoder.get_picture() {
```

Comment:
This match could be simplified to:
```rust
match decoder.get_picture() {
    Err(dav1d::Error::Again) => { ... }
    r => return r.map_err(error_map),
}
```

Reply:
Applied in bffd087d6a31f2f991e2c0d868000d5c9ebf8dc6